install/installer/pkg/config/v1/config.go

```diff
@@ -152,6 +155,7 @@ type ObjectStorage struct {
 	Azure              *ObjectStorageAzure        `json:"azure,omitempty"`
 	MaximumBackupCount *int                       `json:"maximumBackupCount,omitempty"`
 	BlobQuota          *int64                     `json:"blobQuota,omitempty"`
+	Resources          Resources                  `json:"resources" validate:"required"`
```

Comment:
I'm a bit hesitant to make this required as that's a breaking change (as far as I know). Can we make it optional instead?

Reply:
I realise there might have been some context missing about the installer, please see this comment where I elaborate a bit more ☺️ https://github.com/gitpod-io/gitpod/pull/10191#pullrequestreview-982693345

Reply:
By default `minio` gets 2G of memory, but it's not obvious where it is set ([have to dig to find the original yaml, which gets overwritten eventually](https://github.com/gitpod-io/gitpod/blob/main/install/installer/third_party/charts/minio/values.yaml#L13)) - that's why I opted for making it required as it makes it explicit. I realise I should have left a bit more context.

Reply:
> When you have a default value, you don't need to set a value in the config. The user-given config from the file is merged with the default config in the code.

I thought it would be like that as well, but it seems that the merging is done at some later stage. 

1. Installer inits a default config
2. The custom configs in `installer.ts` get applied
3. Validation runs and fails if it wasn't set if it was marked as required

https://github.com/gitpod-io/gitpod/blob/main/.werft/jobs/build/deploy-to-preview-environment.ts#L290-L292

Reply:
@vulkoingim: Sorry, I don't get it. :innocent: 

With this installer: `eu.gcr.io/gitpod-core-dev/build/installer:aa-minio-resources.20` (where it was required and not `omitempty`), you can run
1. `installer init > config.yaml`
2. Remove
   ```
   resources:
     requests:
       memory: 256Mi
   ```
   from `objectStorage` block in `config.yaml`
3. and `installer validate config --config config.yaml` success as expected!?


The only difference between `omitempty` and `validate:"required"` is whether it is generated by `installer init` or not. And with `omitempty` it's not part of the generated `config.yaml` but I think it should. That makes that config value easier to discover. That's why I would vote for settings is as required and not `omitempty`. What do you think?

Reply:
> The only difference between omitempty and validate:"required" is whether it is generated by installer init or not. And with omitempty it's not part of the generated config.yaml

Sorry for leading you astray @vulkoingim - I thought that adding `validate:"required"` meant it had to be part of config.yaml.

Reply:
@corneliusludmann 
> The only difference between omitempty and validate:"required" is whether it is generated by installer init or not. And with omitempty it's not part of the generated config.yaml but I think it should.

The `required` part is also used for validation, not only for generating the config. So it will be generated, but without a default value (or overwritten in `installer.ts`) it will be set to null, so eventually it will fail the validation. It would look like this:

```
objectStorage:
  inCluster: true
  resources:
    requests: null
```

In general I also think that we should set it to `required` and give it a default value in the [config.go:53](https://github.com/gitpod-io/gitpod/pull/10158/files#diff-ed0dab0d7e4e73c6487f629ed17403feeb8ee3b8cf1601e5c940f7a9365ccf17R53), as I've done.

But what Mads also says is true - if we set it to `required` and someone is using an old config with this new version of the installer, where this part is not generated - it will fail.

```
{
  "valid": false,
  "fatal": [
    "Field 'Config.ObjectStorage.Resources.Requests' is required"
  ]
}

```

Reply:
I just tested the behavior for both versions and both fail for this config:
```
objectStorage:
  inCluster: true
  resources:
    requests: null
```

However, when the new config value `resources` in `objectStorage` is missing entirely (that is what old config files would have, it works with both versions as well.

And: Both versions generate the resources block (I was wrong here and it makes sense since the value is not empty and thus not omitted).

Therefore, I'm fine with the current implementation.